cloudinit/net/network_manager.py

```diff
@@ -173,18 +173,44 @@ def _set_ip_method(self, family, subnet_type):
         self.config[family]["method"] = method
         self._set_default(family, "may-fail", "false")
 
-    def _add_numbered(self, section, key_prefix, value):
-        """
-        Adds a numbered property, such as address<n> or route<n>, ensuring
-        the appropriate value gets used for <n>.
-        """
+    def _get_next_numbered_section(self, section, key_prefix) -> str:
         if not self.config.has_section(section):
             self.config[section] = {}
         for index in itertools.count(1):
             key = f"{key_prefix}{index}"
             if not self.config.has_option(section, key):
-                self.config[section][key] = value
-                break
+                return key
+        return "not_possible"  # for typing
+
+    def _add_numbered(self, section, key_prefix, value):
+        """
+        Adds a numbered property, such as address<n> or route<n>, ensuring
+        the appropriate value gets used for <n>.
+        """
+        key = self._get_next_numbered_section(section, key_prefix)
+        self.config[section][key] = value
+
+    def _add_route_options(self, section, route, key, value):
+        """Add route options to a given route
+
+        Example:
+        Given:
+          section: ipv4
+          route: route0
+          key: mtu
+          value: 500
+
+        Create line under [ipv4] section:
+            route0_options=mtu=500
```

Comment:
I had to dig a little to find docs that reference this syntax, but [I found them](https://networkmanager.pages.freedesktop.org/NetworkManager/NetworkManager/nm-settings-keyfile.html). Do you think we should add a link to the function docstring above?

Reply:
Most of this file is for generating these keyfiles. I'll add a more general reference, but I think it'd be odd to put it on this function specifically.